cmd/server/main.go

```diff
@@ -141,3 +145,21 @@ func main() {
 	log.Printf("Listening on :%v", config.Port)
 	log.Fatal(srv.ListenAndServe())
 }
+
+func userEmailKeyFunc() httplimit.KeyFunc {
+	return func(r *http.Request) (string, error) {
+		ipKeyFunc := httplimit.IPKeyFunc("X-Forwarded-For")
```

Comment:
This is a nitty performance thing. We should move this outside of the return so it's only allocated once. Here it will allocate on each request.